src/main/java/agents/leeroy/FortificationActionSupplier.java

```diff
@@ -51,10 +51,10 @@
                     .collect(Collectors.toSet()));
             nextLevelTerritoryIds.addAll(subseqTerritorySet);
         }
-        if (! levelActions.isEmpty()) {
+        if (! levelActions.isEmpty() || nextLevelTerritoryIds.isEmpty()) {
+            // some found on level or last level (i.e. no fortification possible)
             return levelActions;
```

Comment:
What is `levelActions` if `nextLevelTerritoryIds.isEmpty()`

Reply:
it's an empty HashSet, i.e. no actions possible